torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -3123,6 +3125,76 @@ Error PyTorchModelLoader::loadLogicalNot(const torch::jit::Node *ptNode) {
   return addValueMapping(outputs[0], F_.createNot("logical_not", glowInput));
 }
 
+Error PyTorchModelLoader::loadIndexPut(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+
+  glow::NodeValue input;
+  ASSIGN_VALUE_OR_RETURN_ERR(input, getGlowNodeValueForValue(inputs[0]));
+  std::vector<glow::NodeValue> *indices;
+  ASSIGN_VALUE_OR_RETURN_ERR(
+      indices, iValToNodeValueList(getGlowIValueForValue(inputs[1])));
+  glow::NodeValue value;
+  ASSIGN_VALUE_OR_RETURN_ERR(value, getGlowNodeValueForValue(inputs[2]));
+
+  auto *concatNode = F_.createConcat("concat", *indices, 0);
+  auto *reshapeNode =
+      F_.createReshape("reshape", concatNode->getResult(),
+                       {indices->size(), (*indices)[0].dims()[0]});
+  auto *transposeNode =
+      F_.createTranspose("transpose", reshapeNode->getResult(), {1, 0});
+
+  bool cumulative = false;
+  ASSIGN_VALUE_OR_RETURN_ERR(
+      cumulative,
+      iValToBool(getGlowIValueForValue(ptNode->namedInput("accumulate"))));
+
+  auto expectedValueDims =
+      input.dims().drop_front(transposeNode->getResult().dims()[1]).vec();
+  expectedValueDims.insert(expectedValueDims.begin(),
+                           transposeNode->getResult().dims()[0]);
+
+  const auto dimsProduct =
+      std::accumulate(expectedValueDims.begin() + 1, expectedValueDims.end(), 1,
+                      std::multiplies<unsigned_t>());
+  const auto valueDimsProduct =
+      std::accumulate(value.dims().begin() + 1, value.dims().end(), 1,
+                      std::multiplies<unsigned_t>());
+
+  if (value.dims().size() == 1 && value.dims()[0] == 1) {
```

Comment:
Could you add some comments in this pr to indicate what are you doing with all these indexes?

Reply:
Hi, sorry about the delay. I replaced this whole confusing block of code with 

```
 value = F_.broadcastInputs(/* axis */ -1, {value, valueConstant})[0];
 ```
 
 as that is what it essentially was doing. Let me know if there are more changes needed.